Source/WebKit/UIProcess/XR/openxr/OpenXRLayer.cpp

```diff
@@ -54,17 +71,11 @@ std::optional<PlatformXR::FrameData::ExternalTexture> OpenXRLayer::exportOpenXRT
     glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
     glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
 
-    auto* glContext = WebCore::GLContext::current();
-    ASSERT(glContext);
+    auto image = display.createImage(context.platformContext(), EGL_GL_TEXTURE_2D, (EGLClientBuffer)(uint64_t)openxrTexture, { });
```

Comment:
I see you're moving existing code, but we might consider expressing this as

```suggestion
    auto image = display.createImage(context.platformContext(), EGL_GL_TEXTURE_2D, reinterpret_cast<EGLClientBuffer>(static_cast<uintptr_t>(openxrTexture)), { });
```

I've grep'd the code and this approach is used in ANGLE, bmalloc or JavaScriptCore (where they use `void*` instead of `EGLClientBuffer`, but it'd be semantically the same)

Reply:
I don't know, but I prefer to change this in a separate patch and do the same in AcceleratedSurface.cpp

Reply:
We have also had instances in the past where, no matter what we did with C++ casts, there would be always some odd platform where the code would not compile, and ended up using a plain C cast (for examples, see [263299@main](https://commits.webkit.org/263299@main), or [194561@main](https://commits.webkit.org/194561@main))&mdash;I don't know if that would be the case here, only wanting to point out there are precedents where the most reasonable thing to do was the good ol' C cast.